### ISSUE TITLE
Crashes with invalid GIF data

### ISSUE THREAD

**Issue by qvik** [bahlo/SwiftGif#18]
Hi, trying to load invalid data as a GIF crashes this code with EXC_BAD_ACCESS. This is due to the unsafe code in delayForImageAtIndex(). Proper checks should be in place and the UIImage.gifWithData() should throw / return nil if invalid data was handed in. 

As a quick hack, the following is how I made this 'work' (= not crash):

```swift
    class func delayForImageAtIndex(index: Int, source: CGImageSource!) -> Double {
        var delay = 0.1
        
        guard let cfProperties = CGImageSourceCopyPropertiesAtIndex(source, index, nil) as? NSDictionary,
            gifProperties = cfProperties[kCGImagePropertyGIFDictionary as NSString] as? NSDictionary else {
                return delay
        }

        var delayTime = gifProperties[kCGImagePropertyGIFUnclampedDelayTime as NSString]
        
        if delayTime?.doubleValue == 0.0 {
            delayTime = gifProperties[kCGImagePropertyGIFDelayTime as NSString]
        }
        
        if let delayTime = delayTime {
            delay = delayTime.doubleValue
        }
        
        if delay < 0.1 {
            delay = 0.1 // Make sure they're not too fast
        }
        
        return delay
    }
```

Also, building for the device, I get the linker error:

```
ld: framework not found SwiftGifOrigin for architecture arm64
```

- Matti

**Comment by bahlo** [comment]
Thank you for the information and sample code! :+1:

**Comment by bahlo** [comment]
I tested your code, but the tests are failing due to failed access of `cfProperties`.
I would be very grateful if you could fix this error and create a PR? :100:

**Comment by qvik** [comment]
What do you mean 'failed access of cfProperties'? The code I posted compiles & executes without errors on Xcode 7.x build?

It is fairly crucial for our system that the app does not crash when possibly handed a broken .gif, so having this fix is a must. 

-Matti

**Comment by bahlo** [comment]
Closed via #25, feel free to reopen if the issue persists.
